src/py21cmfast/wrapper.py

```diff
@@ -2673,6 +2681,21 @@ def run_lightcone(
                 regenerate=regenerate,
             )
 
+            if coeval_callback is not None:
+                coeval = Coeval(
+                    redshift=z,
+                    initial_conditions=init_box,
+                    perturbed_field=pf2,
+                    ionized_box=ib2,
+                    brightness_temp=bt2,
+                    ts_box=st2 if flag_options.USE_TS_FLUCT else None,
+                    photon_nonconservation_data=_get_photon_nonconservation_data()
+                    if flag_options.PHOTON_CONS
+                    else None,
+                    _globals=None,
+                )
+                coeval_callback_output.append(coeval_callback(coeval))
```

Comment:
I suggest adding a check here to see whether the given function will work

Reply:
Do you think it is okay just to run a function once? What if it does some irreversible change?

Reply:
I'm not sure what you mean. I would just put an `try, except`...

Reply:
Ah okay. So you would just skip in the case the function doesn't work? Just thinking if maybe it's a good thing that the code fails in the case the function fails...